paparazzi/src/main/java/app/cash/paparazzi/PaparazziSdk.kt

```diff
@@ -165,34 +163,29 @@ public class PaparazziSdk @JvmOverloads constructor(
     renderSession.release()
     bridgeRenderSession.dispose()
     cleanupThread()
-    frameHandler?.close()
-    frameHandler = null
 
     renderer.dumpDelegates()
     logger.assertNoErrors()
   }
 
-  public fun setFrameHandler(handler: FrameHandler) {
-    frameHandler = handler
-  }
-
   public fun <V : View> inflate(@LayoutRes layoutId: Int): V =
     layoutInflater.inflate(layoutId, null) as V
 
-  public fun snapshot(composable: @Composable () -> Unit) {
+  public fun snapshot(frameCallback: (BufferedImage) -> Unit, composable: @Composable () -> Unit) {
```

Comment:
I'm not sure that passing the frame callback to every snapshot/gif every method is ideal.

preferred:
```
val frameCallback = {...}

sdk = SDK(frameCallback)
// or
// sdk.setFrameCallback(frameCallback)
sdk.setup()
sdk.snapshot(view, ...)
// update view model
sdk.snapshot(view, ...)
sdk.shutdown()
```

vs:
```
val frameCallback = {...}

sdk = SDK()
sdk.setup()
sdk.snapshot(view, ..., frameCallback)
// update view model
sdk.snapshot(view, ..., frameCallback) // <== why pass this again?
sdk.shutdown()
```

Now I realize it's also not ideal to pass the frame callback via a setter vs a constructor argument, but this is primarily because the current use case is the Junit Rule, i.e., the frame handler takes the name of the method which is only known after the rule (and therefore the sdk) is created, thus the deferred state management.

I *do* like moving `handler.use` out of the SDK, but it looks like:
* leaving the field variable
* keeping the IllegalStateException check in takeSnapshots
* removing `frameHandler?.close()`

fixes the double close problem.

Reply:
> preferred:
> 
> ```
> val frameCallback = {...}
> 
> sdk = SDK(frameCallback)
> // or
> // sdk.setFrameCallback(frameCallback)
> sdk.setup()
> sdk.snapshot(view, ...)
> // update view model
> sdk.snapshot(view, ...)
> sdk.shutdown()
> ```

Took a stab at the preferred path, lmk what you think @jrodbx

Reply:
nice!